server/plugin.go

```diff
@@ -484,6 +482,22 @@ func (p *Plugin) onActivate() error {
 		return err
 	}
 
+	p.remoteID = "msteams"
+	if !p.getConfiguration().DisableSyncMsg {
```

Comment:
Can you confirm if this check is needed? I would like to always register the SharedChannels for the plugin.

Reply:
@wiggin77 :point_up:

Reply:
I think we want an easy way to to turn this off in case of a problem in production.  However we don't want to use a fake remoteID either since that may be hard to clean up later.   I would suggest calling `RegisterPluginForSharedChannels` always, but then calling `UnregisterPluginForSharedChannels` if `p.getConfiguration().DisableSyncMsg == true`.  That way you will have the real remoteID.

Reply:
My concern is the remoteID changing over disabling/enabling. Is the remote id going to be stable or it is going to change every time that I unregister the plugin?

Reply:
Anyway, I applied the changes that you proposed.

Reply:
@jespino  You're right.  I should soft-delete the record on Unregister and clear the DeleteAt on register... or derive the remoteID from the pluginID.  Either way we want the same remoteID for every unique plugin ID.  Any preference?  I think I lend toward hashing the pluginID to make the remoteID.

Reply:
Sounds good enough, but also slightly more error prone. I don't think is a big deal.

Reply:
> Either way we want the same remoteID for every unique plugin ID.

This is probably fine for our current purposes, but thinking (very) long term, I could also see us supporting multiple Microsoft tenants concurrently which might necessitate a distinct `remoteID` for each (if we wanted to, for example, distinguish each in the UI?)

Stepping back, I'm also just wanting to confirm that /we/ (the plugin) don't really care about the value here at all, and that it's just an artifact of the way the shared channels subsystem works with users that happen to have a `RemoteID` set. In other words, we want all DMs and all GMs for all users, regardless of whether anyone has a `RemoteID` set. Does that remain accurate @wiggin77?

Reply:
If you set the appropriate flag when registering then yes, the MS Teams plugin will get sync for all DM/GMs regardless of remoteID value.  Other plugins using the API will likely not set that flag and the remoteID on posts/users/etc will matter.  

Should we modify the register API to accept a remoteID, thus allowing a plugin to call register for each tenant?  That would also leave it up to the plugin to ensure idempotency.   cc @jespino @lieut-data

Reply:
> Should we modify the register API to accept a remoteID, thus allowing a plugin to call register for each tenant? That would also leave it up to the plugin to ensure idempotency. cc @jespino @lieut-data

On the one hand, I don't want to preemptively build for the future. But I do like the idea of giving the plugin control over the "primary key" backing the cursor based model. At the moment, it feels just a bit too decoupled. So I'm in favour of the idea. Is the `RemoteId` constrained by some length we should remain aware of? (i.e. is it long enough for a UUID?)

Reply:
It's the same length as most IDs in MM.